Add docker setups

lazyguru opened the pull request:
## Intro
This repo will hold docker-compose.yml files to run Sublinks for both demo and local dev purposes (eventually, production/live too). This PR is the first (major) step towards that goal.

### CURRENT STATUS
Please note, this repo currently holds configuration files intended for development purposes only. As we get closer to our initial MVP release we will provide sample files for use in a live site.

### Demo Site

- Run `docker-compose -f docker-compose.demo.yml up [-d]`

### Frontend Dev

- Run `docker-compose -f docker-compose.frontend.yml up [-d]`

NOTE: This needs to be run from a subdirectory of the frontend repo. A separate PR will be opened to replace the current `etc/dev` directory with a `docker` directory containing this repo as a submodule

lazyguru committed:
Add initial README.md

lazyguru committed:
Add common services

lazyguru committed:
Add demo config

lazyguru committed:
Add backend service

lazyguru committed:
Add Lemmy UI

lazyguru committed:
Add federation

lazyguru committed:
Add frontend configs

kgilles reviewed (commented):
Things look good to me after a first pass. Although I can't say I'm familiar with most things outside the frontend setup. Which did get going and was accessible for me after pointing the app volume to my local frontend root.

Only thing for now is that you definitely want to ignore node_modules/ for the seeder. That's about 600 files that don't need to be committed.

lazyguru commented:
> Things look good to me after a first pass. Although I can't say I'm familiar with most things outside the frontend setup. Which did get going and was accessible for me after pointing the app volume to my local frontend root.

Cool. I'm targeting frontend repo first anyway so was hoping it would all be working. I'm intending on introducing a "docker" folder into each repo as a submodule.  That way the instructions would be more or less the same for each repo (eg, `cd docker && docker-compose -f docker-compose.<area>.yml up`)

> Only thing for now is that you definitely want to ignore node_modules/ for the seeder. That's about 600 files that don't need to be committed.

Good catch. I just pushed a commit removing it

lazyguru committed:
Add codeowners file

lazyguru commented:
Moving to ready for review so I can target integration into frontend repo. I will open new PRs for backend and federation configs

lazyguru commented on .github/CODEOWNERS:
NOTE: I added a codeowners file and setup repo permissions such that based on the paths defined core-developers AND the respective team will be auto-added to any PR that modifies a file in that path

lazyguru commented on lemmy-ui/services.yml:
```diff
@@ -0,0 +1,30 @@
+    image: dessalines/lemmy-ui:0.19.3
+    environment:
+      #- LEMMY_UI_DEBUG=true
+      - LEMMY_UI_HOST=0.0.0.0:3000
```
NOTE: I changed from the default 1234 to 3000. This was to match the sublinks-ui and so I could reference either container as "ui" for purposes of the nginx config files. Only one of the 2 containers can be run (but not sure why you would ever need to run both?)

lazyguru committed:
Expose queue port externally

kgilles commented on README.md:
```diff
@@ -1,2 +1,22 @@
+
+## Frontend Dev
+
+Run `docker-compose -f docker-compose.frontend.yml up [-d]`
```
`[-d]` aborts the command for me with `no matches found: [-d]`.
Command succeeds if I use `-d` without the brackets.

lazyguru replied:
The brackets indicate optional command arguments. I'll add a comment below each command to clarify

kgilles replied:
TIL

lazyguru replied:
I updated the README. Please let me know if that helps?

kgilles commented on common/seeder/package.json:
```diff
@@ -0,0 +1,10 @@
+    "seed": "node ./job-seeder.js"
+  },
+  "dependencies": {
+    "sublinks-js-client": "^0.0.3"
```
Version 0.0.13 is the latest one. This could be a good opportunity to upgrade the seeder dependency.

lazyguru replied:
Updated. Also added Dependabot config to (hopefully) keep it updated

kgilles reviewed (changes requested):
Couple of small things noted.

Would it be easy enough to add some kind of option/configuration to allow specifying the platform for the Sublinks API and Federation images. As I mentioned in the Fed chat I can't run this as-is. I need to set the platform versions since these two images aren't pushed with a matching linux arch.

lazyguru commented:
I'm fixing that in the individual repos (to build and push multi-arch images). Already have a PR open in federation

lazyguru committed:
Add explanation about optional parameters to README

lazyguru committed:
Update sublinks-js-client

Also add node_modules to dockerignore file to avoid it being added in
case FE team is working on seeder

lazyguru committed:
Add dependabot config

kgilles reviewed (approved):
Beautiful
